File: DAY2.py
def find_missing_by_sum(arr):
    n = len(arr) + 1
    expected_sum = n * (n + 1) // 2
    actual_sum = sum(arr)
    return expected_sum - actual_sum

def find_missing_by_xor(arr):
    n = len(arr) + 1
    xor_res = 0
    for i in range(1, n + 1):
        xor_res ^= i
    for num in arr:
        xor_res ^= num
    return xor_res

arr = [1, 2, 4, 5]
print(find_missing_by_sum(arr))
print(find_missing_by_xor(arr))

File: DAY2.java
public class MissingNumberFinder {
    public static int findMissingBySum(int[] arr) {
        int n = arr.length + 1;
        long expectedSum = (long) n * (n + 1) / 2;
        long actualSum = 0;
        for (int num : arr) {
            actualSum += num;
        }
        return (int) (expectedSum - actualSum);
    }

    public static int findMissingByXOR(int[] arr) {
        int n = arr.length + 1;
        int xorResult = 0;
        for (int i = 1; i <= n; i++) {
            xorResult ^= i;
        }
        for (int num : arr) {
            xorResult ^= num;
        }
        return xorResult;
    }

    public static void main(String[] args) {
        int[] arr = {1, 2, 4, 5};
        System.out.println(findMissingBySum(arr));
        System.out.println(findMissingByXOR(arr));
    }
}
